modules/apps/product-navigation/product-navigation-user-personal-bar-web/src/main/resources/META-INF/resources/view.jsp

```diff
@@ -57,5 +57,100 @@
 		<span class="sign-in text-default" role="presentation">
 			<aui:icon cssClass="sign-in text-default" data="<%= anchorData %>" image="user" label="sign-in" markupView="lexicon" url="<%= themeDisplay.getURLSignIn() %>" />
 		</span>
+
+		<aui:script sandbox="<%= true %>">
+			var signInLink = document.querySelector('.sign-in > a');
+
+			if (signInLink && signInLink.dataset.redirect === 'false') {
+				var signInURL = '<%= themeDisplay.getURLSignIn() %>';
+
+				var modalSignInURL = Liferay.Util.addParams(
+					'windowState=exclusive',
+					signInURL
+				);
+
+				var setModalContent = function (html) {
+					var modalBody = document.querySelector('.liferay-modal-body');
+
+					if (modalBody) {
+						var fragment = document
+							.createRange()
+							.createContextualFragment(html);
+
+						modalBody.innerHTML = '';
+
+						modalBody.appendChild(fragment);
+					}
+				};
+
+				var loading = false;
+				var redirect = false;
+				var html = '';
+				var modalOpen = false;
+
+				var fetchModalSignIn = function () {
+					if (loading || html) {
+						return;
+					}
+
+					loading = true;
+
+					Liferay.Util.fetch(modalSignInURL)
+						.then(function (response) {
+							return response.text();
+						})
+						.then(function (response) {
+							loading = false;
```

Comment:
So if fetch starts, then we close, our `onClose()` handler resets the variables (eg. `loading = false`, `html = ''` etc).

It seems that we should bail here in that case:

```js
if (!loading) {
    return;
}

loading = false;
```

Otherwise, we'll end up setting `html` and possibly `response` too. It sounds harmless enough, probably, but it seems better to be safe than sorry. For example, if we don't bail early, and then we open the modal again, we'll hit:

```js
if (loading || html) {
    return;
}
```

Which seems to me like we'd return early due to truthy `html`, and then we'd end up rendering "stale" HTML.

Reply:
I added a commit with the change.

Note - if you open modal > close modal > open modal, all before fetch is done, this will not trigger and the modal content will load normally. That's ok, no caching is involved.